### DISCUSSION TITLE
motion consumes too much memory on my Raspberry PI

### DISCUSSION THREAD

**Discussion by angryferret** [Motion-Project/motion#1629] [General]
I am running motion Version 4.5.0on a Raspberry Pi 3 Model B 

```
root@netcam:~# inxi 
CPU: Quad Core Model N/A (-MCP-) speed/min/max: 1200/600/1200 MHz Kernel: 5.15.61-v8+ aarch64 Up: 1d 23h 38m 
Mem: 892.9/985.6 MiB (90.6%) Storage: 29.81 GiB (18.1% used) Procs: 127 Shell: Bash inxi: 3.3.01 

```

The motion process is frequently crashing, killed by the kernel with OOM errors. 

```

top - 09:33:54 up 1 day, 23:39,  2 users,  load average: 0.60, 0.52, 0.52
Tasks: 128 total,   1 running, 127 sleeping,   0 stopped,   0 zombie
%Cpu(s): 13.9 us,  0.7 sy,  0.0 ni, 85.4 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
MiB Mem :    909.6 total,     44.7 free,    749.3 used,    115.6 buff/cache
MiB Swap:    100.0 total,     71.0 free,     29.0 used.    107.2 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                               
  26702 motion    20   0 4178388 567568    144 S  55.1  60.9 629:14.72 motion
```         

Are there any config settings I can use to get around this problem? The PI is just for motion only, no other purpose. 
Is the PI not specced high enough to run motion?

**Comment by Mr-Dave** [comment]
It depends.  I have a RPI 3 and it does run. The size of the image has an exponential effect on CPU usage and requirements.  i.e. Double the image size probably quadruples the usage.  (Others can correct the exact multiple but this is how everyone should think about it.).  I will say that I did Valgrind on 4.5.0 on the PI and did not see any leaks.  My first recommendations would be to lower resolution if it is high and possibly start with a fresh image of Raspbian.

**Comment by tosiara** [comment]
61% of 900MB RAM looks too much
An example of an ARM device running 720p USB camera:

```
MiB Mem :    967.2 total,     45.1 free,    146.8 used,    775.3 buff/cache
MiB Swap:    483.6 total,    473.9 free,      9.8 used.    787.3 avail Mem 

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                                               
15827 motion    20   0  256736  37824   7852 S  50.0   3.8   2293:51 motion 
```
